### DISCUSSION TITLE
Replace date regex in new journal tiddler body and set field

### DISCUSSION THREAD

**Discussion by EHJ-52n** [TiddlyWiki/TiddlyWiki5#6576] [Q&A]
Dear Community,
I am using tiddlywiki on a daily basis since 2018. I create a daily journal tiddler that collects and lists tiddlers on a date based field and other by their creation date. Only the title of the tiddler is created based on the current date. The other filter expression need to be updated/adjusted every day by hand. I would like to improve this, so all date specific filter expressions and the caption field are filled automatically. In addition, a tag should be applied based on the date information, too.

The "source" of the journal tiddler looks like the following for today, 31.03.2022:

* **Title**: 2022-03-31
* **Tag**: 03_2022
* **Field::caption**: 31.

**Body**:
```
<$details summary="Today's Tasks">

''Open''

<$list filter="[!has[draft.of]tag[task]journalday[2022-03-31]!tag[done]sort[created]]">

<$checkbox tag="done"> <$link to={{!!title}}><$view field="title"/></$link></$checkbox>

</$list>

''Done''

<$list filter="[!has[draft.of]tag[task]tag[done]journalday[2022-03-31]sort[created]]">

<$checkbox tag="done"> ~~<$link to={{!!title}}><$view field="title"/></$link>~~</$checkbox>

</$list>

</$details>
<$details summary="Tiddler">

<<list-links "[regexp:created[^20220331]!is[system]]">>

</$details>


!!! Journal

<$list filter="[!has[draft.of]journalday[2022-03-31]sort[title]]">
<$transclude field="text" mode="block"/>
</$list>
```
I am using the field `journalday` because I do not want to have to many technical tags in my system. 13 per year is okay, but not hundreds (e.g. one tag per workday).
Do you have any recommendations where to start and how to implement this?

I am a developer, but not a JavaScript (and Tiddlywiki) one.

**Comment by pmario** [comment]
Hi, IMO the format of the task tiddler is missing. ... 

From the description I'm not 100% sure what you try to achieve, but I think the whole "list / filter creation" process should work without human intervention. right?

Just brainstorming

There may be a Journal-dashboard tiddler that works with a macro and always shows the current day. This dashboard can change dynamically. ... 

But at some time in the day you may need to click a button and create an "archive journal", where the date is "hardcoded", so it also can be accessed at a later date. 

If the journals contain "meta data", they have to exist. If journals are purely dynamic a date selector in the dashboard would be enough to see "older" journals. .. IMO it depends on your taste.

If you want to avoid technical tags, you can use "technical fieldl". I think at the beginning technical tags shouldn't be avoided, because that also avoids to see "patterns". If a pattern evolves over time, there is a very high chance, that it can be "abstracted" away with some code or templates. ... Not seeing those patterns usually makes things more complex. 

just some thoughts

**Reply by pmario** [reply]
Where is the `field::caption` field used?

**Reply by EHJ-52n** [reply]
In the contents tab on the right.
![image](https://user-images.githubusercontent.com/3839896/161708275-dfc0866c-f87e-4c9a-9caa-f37c902bc083.png)

**Comment by Jermolene** [comment]
Hi @EHJ-52n it might be worth asking this over at https://talk.tiddlywiki.org/ where there's a larger audience of very ingenious people who might be able to help.

**Reply by EHJ-52n** [reply]
@Jermolene any category recommendation for my posting @ talk.tiddlywiki.org?

**Reply by saqimtiaz** [reply]
Discussion category.
